mlrose/algorithms.py

```diff
@@ -87,7 +87,10 @@ def hill_climb(problem, max_iters=np.inf, restarts=0, init_state=None,
             problem.find_neighbors()
             next_state = problem.best_neighbor()
             next_fitness = problem.eval_fitness(next_state)
-
+            
+            if curve:
+                fitness_curve.append(problem.get_fitness())
+            
             # If best neighbor is an improvement, move to that state
             if next_fitness > problem.get_fitness():
```

Comment:
@gkhayes , When you say "wrong location" are you referring to the fact that it's happening before this line? If so, that makes sense. 

I believe we both agree on that fact that it should be within the while loop, correct?

Reply:
Sorry, I should have been clearer. Anyway, what you have done is in agreement with my thinking.